src/cryptography/hazmat/primitives/kdf/kbkdf.py

```diff
@@ -27,10 +37,9 @@ class CounterLocation(utils.Enum):
     AfterFixed = "after_fixed"
 
 
-class KBKDFHMAC(KeyDerivationFunction):
+class _KBKDFBase(KeyDerivationFunction):
```

Comment:
We strongly attempt to avoid subclassing and inheritance (we had none at all until mypy forced us to stop using ABC registration). There's a lot shared here, but I think we should avoid calls to `super()`.

Let's move the shared logic from this into functions and then have two separate classes `KBKDFHMAC(KeyDerivationFunction)` and `KBKDFCMAC(KeyDerivationFunction)` that call those functions to reduce duplication.

There will be some duplication in `__init__` with this approach, but overall I'd prefer it.

Reply:
Ok, I'll rewrite it as you suggest. I have some other stuff to handle next week but after that I can get back to this PR. If you have any other comments please share them.

Reply:
I'm okay with this approach.